http/src/request/application/set_command_permissions.rs

```diff
@@ -8,29 +8,156 @@ use crate::{
     response::ResponseFuture,
     routing::Route,
 };
-use serde::{Serialize, Serializer};
+use serde::{ser::SerializeSeq, Serialize, Serializer};
 use twilight_model::{
     application::command::permissions::CommandPermissions,
     id::{ApplicationId, CommandId, GuildId},
 };
 
-#[derive(Serialize)]
-struct GuildCommandPermission<'a> {
-    id: &'a CommandId,
-    permissions: &'a CommandPermissions,
+#[derive(Clone, Copy, Debug)]
+struct OptionalCommandPermissions<'a>(
+    [Option<&'a CommandPermissions>; InteractionError::GUILD_COMMAND_PERMISSION_LIMIT],
+);
+
+impl OptionalCommandPermissions<'_> {
+    /// Create a new list of command permissions with `None` elements.
+    const fn new() -> Self {
+        Self([None; InteractionError::GUILD_COMMAND_PERMISSION_LIMIT])
+    }
+
+    /// Determine the number of elements present.
+    ///
+    /// If all elements are present then
+    /// [`InteractionError::GUILD_COMMAND_PERMISSION_LIMIT`] is returned.
+    ///
+    /// If no elements are present then 0 is returned.
+    fn amount_present(&self) -> usize {
+        // Iterate over the elements until we find one that is None. If we don't,
+        // then the maximum number are present.
+        self.0
+            .iter()
+            .position(Option::is_none)
+            .unwrap_or(InteractionError::GUILD_COMMAND_PERMISSION_LIMIT)
+    }
+}
+
+impl Serialize for OptionalCommandPermissions<'_> {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut seq = serializer.serialize_seq(Some(self.amount_present()))?;
+
+        for maybe_value in &self.0 {
+            if let Some(value) = maybe_value.as_ref() {
```

Comment:
Nit: A match might be cleaner here and save a few lines.

Reply:
Even better I `while let`'d it